ext/nokogiri/extconf.rb

```diff
@@ -563,18 +580,19 @@ def install
   "exslt" => ['exsltFuncRegister',      'libexslt/exslt.h'],
 }.each { |lib, (func, header)|
   have_func(func, header) ||
-  have_library(lib, func, header) ||
+  have_library(lib, func, header, iconv_libs) ||
```

Comment:
Also, isn't there `pkg-config` in the PATH?  If you omit --with-xml2-* flags, extconf.rb should call pkg-config to read appropriate flags from the libxml-2.0 package configuration.

```ruby
  dir_config('xml2').any?  or pkg_config('libxml-2.0')
  dir_config('xslt').any?  or pkg_config('libxslt')
  dir_config('exslt').any? or pkg_config('libexslt')
```